rasa_sdk/knowledge_base/utils.py

```diff
@@ -162,3 +163,29 @@ def reset_attribute_slots(
             slots.append(SlotSet(attr, None))
 
     return slots
+
+
+async def get_object_type_dynamic(
+    tracker: "Tracker",
+    object_types: List,
+) -> Optional[Text]:
+    """
+    If the user ask a question about an attribute using an object name and
+    without specifying the object type, then this function searches the
+    corresponding object type. (e.g: when user asks'price range of B&B', this
+    function detects the object type as 'hotel')
+
+    Args:
+        tracker: the tracker
+        object_types: list of object types in the knowledge base
+
+    Returns: the name of the object type
+    """
+    entities = tracker.latest_message["entities"]
+    entities_values = [entities[i]["entity"] for i in range(len(entities))]
+    for entity in entities_values:
+        if entity in object_types:
+            object_type_dynamic = entity
+            return object_type_dynamic
```

Comment:
We, too, had this concern in the beginning. But for the proposed solution, the assumption is that the user asks only about one object_type. We made this assumption based on rasa’s current implementation logic of knowledge base actions and the [documentation examples](https://rasa.com/docs/rasa/action-server/knowledge-bases/), which only demonstrate querying with a single object_type entity. In the existing knowledge base action logic, the users cannot inquire about multiple object types as the slot identifies only the latest object type in the user message and queries accordingly. We can solve this by changing the object_type slot to a list and re-writing the core components of the knowledge base action logic. However, this would be a major change. That is why we limited our scope in this PR.